frame/staking/src/lib.rs

```diff
@@ -352,6 +351,9 @@ type NegativeImbalanceOf<T> = <<T as Config>::Currency as Currency<
 	<T as frame_system::Config>::AccountId,
 >>::NegativeImbalance;
 
+type AccountIdOf<T> = <T as frame_system::Config>::AccountId;
+type VotingDataOf<T> = (AccountIdOf<T>, VoteWeight, Vec<AccountIdOf<T>>);
```

Comment:
Lets avoid tuples, and use a struct instead.

Reply:
Agree in principle, but that would end up being a very widespread change across several pallets. This PR is big already; I'd prefer not to add that change here.

Reply:
Created https://github.com/paritytech/substrate/issues/9152 so we don't lose track of the issue.

Reply:
Note that this is IMO only reasonable to do later since going from tuple to struct is not storage-breaking change. Otherwise, doing a migration for just this would have been a PITA.